fix(language-service): correctly handle host directive inputs/outputs

crisbeto opened the pull request:
Adds some logic to correctly handle hidden or aliased inputs/outputs in the language service.

Fixes #48102.

crisbeto commented on packages/compiler-cli/src/ngtsc/typecheck/src/template_symbol_builder.ts:
```diff
@@ -118,38 +119,78 @@ export class SymbolBuilder {
+          tsSymbol: symbol.tsSymbol,
+          exposedInputs: current.inputs,
+          exposedOutputs: current.outputs,
+          selector: meta.selector || '',
```
The `|| ''` here is weird, but I don't know the context behind why the language service requires a selector for all directives, even though directives can be declared without selectors.

atscott replied:
Before the directive composition, a directive couldn't match anything unless it had a selector. I think the correct thing would be to make this optional now. Can you add a TODO or make the change if it's easy enough?

crisbeto replied:
I left a TODO for myself. I'll send a follow-up PR once this one goes in.

atscott reviewed (approved):
👍 LGTM!

crisbeto committed:
fix(language-service): correctly handle host directive inputs/outputs

Adds some logic to correctly handle hidden or aliased inputs/outputs in the language service.

Fixes #48102.

dylhunn commented:
This PR was merged into the repository by commit fd2eea59613ab3cdde871046b6086216d77a386e.
